src/Api/Rest/Rest.php

```diff
@@ -0,0 +1,225 @@
+
+use Joomla\Webservices\Api\ApiBase;
+use Joomla\Webservices\Resource\Item;
+use Joomla\Webservices\Resource\Resource;
+use Joomla\Webservices\Webservices\Webservice;
+use Joomla\Webservices\Webservices\Factory;
+
+use Joomla\DI\Container;
+use Joomla\Event\Event;
+use Joomla\Event\EventImmutable;
+use Joomla\Registry\Registry;
+
+/**
+ * Class to represent a REST interaction style.
+ *
+ * @since  1.2
+ */
+class Rest extends ApiBase
+{
+	/**
+	 * @var    string  Name of the Api
+	 * @since  1.2
+	 */
+	public $apiName = 'rest';
+
+	/**
+	 * Profile object.
+	 * 
+	 * @var  Profile
+	 */
+	private $profile = null;
+
+	/**
+	 * Main resource object
+	 * @var  Resource
+	 */
+	public $resource = null;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param   Container  $container  The Dependency Injection Container object.
+	 * @param   Registry   $options    Optional custom options to load.
+	 *
+	 * @throws  \Exception
+	 * 
+	 * @since   __DEPLOY_VERSION__
+	 */
+	public function __construct(Container $container, Registry $options)
+	{
+		parent::__construct($container, $options);
+
+		// Set initial status code.
+		$this->setStatusCode(200);
+	}
+
+	/**
+	 * Execute the Api operation.
+	 *
+	 * @return  mixed  Webservice object with information on success, boolean false on failure.
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 * @throws  \Exception
+	 */
+	public function execute()
+	{
+		$options	= $this->getOptions();
+
+		$method		= $options->get('method', 'GET');
+		$task		= $options->get('task', '');
+		$format		= $options->get('format', '');
+		$clientName	= $options->get('webserviceClient');
+		$version	= $options->get('webserviceVersion');
+		$resourceName = $options->get('optionName');
+
+		// Map HTTP methods to service operations.
+		switch (strtolower($method))
+		{
+			case 'get':
+				$operation = !empty($task) ? 'task' : 'read';
+				break;
+
+			case 'put':
+				$operation = 'update';
+				break;
+
+			case 'post':
+				$operation = !empty($task) ? 'task' : 'create';
+				break;
+
+			case 'patch':
+				$operation = 'patch';
+				break;
+
+			case 'delete':
+				$operation = 'delete';
+				break;
+		}
+
+		$this->setOption('operation', $operation);
+
+		// If task is pointing to some other operation like apply, update or delete.
+//		if (!empty($task) && !empty($this->configuration->operations->task->{$task}['useOperation']))
+//		{
+//			$useOperation = strtoupper((string) $this->configuration->operations->task->{$task}['useOperation']);
+//
+//			if (in_array($useOperation, array('create', 'read', 'update', 'delete', 'documentation')))
+//			{
+//				$operation = $useOperation;
+//			}
+//		}
+
+		// Get the Profile object for the webservice requested.
+		$this->profile = Factory::getProfile($this->getContainer(), $clientName, $resourceName, $version, $operation);
+
+		$this->webservice = Factory::getWebservice($this->getContainer(), $operation, $this->getOptions());
+
+		// Set initial status code to OK.
+		$this->setStatusCode(200);
```

Comment:
Other than because it's like this in the code at the moment is there any reason we are doing this again (as we already did it once in the constructor)